storage/innobase/handler/handler0alter.cc

```diff
@@ -8433,6 +8433,63 @@ innobase_update_foreign_cache(
 	DBUG_RETURN(err);
 }
 
+/** Changes WITH/WITHOUT SYSTEM VERSIONING for fields.
+@param ha_alter_info Data used during in-place alter
+@param ctx In-place ALTER TABLE context
+@param table MySQL table as it is before the ALTER operation */
+static
+void
+innobase_change_fields_versioning(
+	Alter_inplace_info*		ha_alter_info,
+	ha_innobase_inplace_ctx*	ctx,
+	const TABLE*			table)
+{
+	DBUG_ENTER("innobase_change_fields_versioning");
+
+	DBUG_ASSERT(ha_alter_info);
+	DBUG_ASSERT(ctx);
+	DBUG_ASSERT(ha_alter_info->handler_flags & ALTER_COLUMN_UNVERSIONED);
+
+	uint virtual_count = 0;
+
+	for (uint i = 0; i < table->s->fields; i++) {
+		Field* field = table->field[i];
+
+		if (innobase_is_v_fld(field)) {
+			virtual_count++;
+			continue;
+		}
+
+		Create_field* create_field = NULL;
+		List_iterator_fast<Create_field> it(
+		    ha_alter_info->alter_info->create_list);
+		while (Create_field* cf = it++) {
```

Comment:
Could we create the `it` only once outside the `for` loop, and `it.rewind()` here inside the loop?